azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1256,76 +1361,287 @@ private void fillUpdateTimeAndExecId(List<ExecutableFlow> flows,
+              && currentTime - QUEUE_PROCESSOR_WAIT_IN_MS > lastProcessingTime) {
+              // Refresh executor stats to be used by selector
+              refreshExecutors();
+              // process upto a maximum of maxContinousSubmission from queued
+              // flows
+              processQueuedFlows(maxContinousSubmission);
+              lastProcessingTime = currentTime;
+            }
+            wait(QUEUE_PROCESSOR_WAIT_IN_MS);
+          } catch (InterruptedException e) {
+            logger.info("Interrupted. Probably to shut down.");
+          }
+        }
+      }
     }
   }
 
-  @Override
-  public int getExecutableFlows(int projectId, String flowId, int from,
-      int length, List<ExecutableFlow> outputList)
-      throws ExecutorManagerException {
-    List<ExecutableFlow> flows =
-        executorLoader.fetchFlowHistory(projectId, flowId, from, length);
-    outputList.addAll(flows);
-    return executorLoader.fetchNumExecutableFlows(projectId, flowId);
+  private void refreshExecutors() {
+    synchronized (activeExecutors) {
+      // TODO: rest api call to refresh executor stats
+    }
   }
 
-  @Override
-  public List<ExecutableFlow> getExecutableFlows(int projectId, String flowId,
-      int from, int length, Status status) throws ExecutorManagerException {
-    return executorLoader.fetchFlowHistory(projectId, flowId, from, length,
-        status);
+  private void processQueuedFlows(long maxContinousSubmission) {
+    try {
+      Pair<ExecutionReference, ExecutableFlow> runningCandidate;
+      int submissionNum = 0;
+      while (submissionNum < maxContinousSubmission
+        && (runningCandidate = queuedFlows.peek()) != null) {
+        synchronized (runningCandidate) {
+
+          ExecutionReference reference = runningCandidate.getFirst();
+          ExecutableFlow exflow = runningCandidate.getSecond();
+
+          // TODO: use dispatcher
+          Executor choosenExecutor;
+
+          synchronized (activeExecutors) {
+            choosenExecutor = activeExecutors.iterator().next();
+          }
+
+          if (choosenExecutor != null) {
+            queuedFlows.poll();
+            queuedFlowMap.remove(exflow.getExecutionId());
+
+            try {
+              reference.setExecutor(choosenExecutor);
+              executorLoader.assignExecutor(choosenExecutor.getId(),
+                exflow.getExecutionId());
+              // TODO: ADD rest call to do an actual dispatch
+              callExecutorServer(reference, ConnectorParams.EXECUTE_ACTION);
+              runningFlows
+                .put(exflow.getExecutionId(),
+                  new Pair<ExecutionReference, ExecutableFlow>(reference,
+                    exflow));
+            } catch (ExecutorManagerException e) {
+              logger.error("Failed to process queued flow", e);
+              // TODO: allow N errors and re-try
+              finalizeFlows(exflow);
+            }
+          }
+        }
+        submissionNum++;
```

Comment:
nicer place to increase the submissionNum might be in the block where a successful dispatch happened.